### DISCUSSION TITLE
OpenAPI integration

### DISCUSSION THREAD

**Discussion by knightazura** [retypeapp/retype#228] [General]
Hello folks, first of all thank you for Retype devs! You're rock!

Just to the point, is there any plan to integrate OpenAPI into Retype?

Thanks

**Comment by fabriciomurta** [comment]
This was Retype's original goal -- although not OpenAPI specific -- but as far as generate code documentation out of the project's sources.

We have some work over [C#'s documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) and Roslyn (to link related entities, inheritance, and so on).

There were extensive discussion on Typescript language inline docs support too, but we've been focusing on the generic purpose markdown documentation for now. It alone proved to be challenging and demanded enough for months of work.

But we do indeed think on integrating code documentation generation support. Here's a sneak peek at our [MojeeIO emoji library API documentation generated by Retype](https://docs.mojee.io/api/mojeeio/mojee/).

Nevertheless, I think this would make a very welcome feature request, we'd love to see how many people on the community are looking for API docs generation with Retype. That'd be very helpful to guide our next steps.

**Reply by knightazura** [reply]
That was great, especially MojeeIO example. I think it's probably fit my needs for now.

I originally start to move documentation from simple plain markdown into something like Retype and hope there is an app that can integrated with OpenAPI beautifully.

Previously I've made for API documentation through OpenAPI, but somehow it feels something was miss. Glad I found Retype and just in first time to try, I decided to use it 😁

All in all, if Retype has API docs generation regardless it can integrate OpenAPI or own implementation, it would be perfect! 
Where I can create feature request?

**Reply by fabriciomurta** [reply]
You shall see a "create issue from discussion" link to the right here below the "subscribe/unsubscribe" button. We can do it ourselves too but just curious if that's not available for you.

**Reply by knightazura** [reply]
Ohh okay, don't worry. It's available.

I just want to make sure to create it in the right place 👍
